.github/workflows/main.yml

```
@@ -0,0 +1,82 @@
+name: pixlet
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - 'v*'
+
+jobs:
+  build-and-test-release:
+    name: Build and Test Release Artifacts
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+    - name: Install Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: '1.16.12'
+
+    - name: Checkout code
+      uses: actions/checkout@v2
+
+    - name: Install Linux dependencies
+      if: matrix.os == 'ubuntu-latest'
+      run: sudo ./scripts/setup-linux.sh
+
+    - name: Install macOS dependencies
+      if: matrix.os == 'macos-latest'
+      run: ./scripts/setup-macos.sh
+
+    - name: Build
+      run: make build
+
+    - name: Test
+      run: make test
+
+    - name: Build Release Linux
+      if: matrix.os == 'ubuntu-latest'
+      run: make release-linux
+
+    - name: Build Release macOS
+      if: matrix.os == 'macos-latest'
+      run: make release-macos
+
+    - name: Upload Release Artifacts
+      uses: actions/upload-artifact@v2
+      with:
+        name: release-artifacts
+        path: build
+
+  create-release:
+    name: Create Github Release
+    runs-on: ubuntu-latest
```

```suggestion
    runs-on: ubuntu-latest
    environment: release
```

Need to run in the release environment to access the `TIDBYT_GITHUB_TOKEN` secret.

https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment#using-an-environment